app/jobs/populate_register_data_in_db_job.rb

```diff
@@ -1,11 +1,24 @@
 class PopulateRegisterDataInDbJob < ApplicationJob
   queue_as :default
 
+  module Exceptions
+    class FrontendInvalidRegisterError < StandardError; end
+  end
+
   def perform(register)
     Delayed::Worker.logger.info("Updating #{register.name} in database")
+    begin
     # Initialize client and download / update data
-    register_client = @@registers_client.get_register(register.name.parameterize, register.register_phase.downcase, PostgresDataStore.new(register))
-    register_client.refresh_data
+      register_client = @@registers_client.get_register(register.name.parameterize, register.register_phase.downcase, PostgresDataStore.new(register))
+      register_client.refresh_data
+    rescue InvalidRegisterError
+      # If register data is invalid we want to delete existing entries and records to force a full reload
+      ActiveRecord::Base.transaction do
+        Record.where(register_id: register.id).destroy_all
```

Comment:
update to `Record.find(register.id).destroy_all` ?

Reply:
This does not work as `find` selects one record and `destroy_all` works on multiple.